jobs/scheduled/enqueue_reminders.rb

```diff
@@ -17,14 +17,19 @@ def skip_enqueue?
 
     def user_ids
       interval = SiteSetting.remind_assigns_frequency
+      allowed_groups = SiteSetting.assign_allowed_on_groups.split('|')
 
       TopicCustomField
         .joins(<<~SQL
           LEFT OUTER JOIN user_custom_fields ON topic_custom_fields.value::INT = user_custom_fields.user_id
           AND user_custom_fields.name = '#{PendingAssignsReminder::REMINDED_AT}'
         SQL
         ).joins("INNER JOIN users ON topic_custom_fields.value::INT = users.id")
-        .where("users.moderator OR users.admin")
+        .where('users.id IN (
```

Comment:
Let's do a `JOIN` here instead. `IN` clauses are generally quite inefficient because it becomes an array comparison instead of a potential index scan.

Reply:
I changed it to a join but I'll fetch allowed group ids in another query, don't think it's a major performance hit.